---
float64 velocity
int32 exception_code
